src/settings/dto/create-setting.dto.ts

```diff
@@ -0,0 +1,15 @@
+import { IsNotEmpty } from 'class-validator';
+
+export class CreateSettingDto {
+  @IsNotEmpty()
+  name: string;
+
+  @IsNotEmpty()
+  data_type: string;
```

Comment:
Use @IsEnum with the list of allowed values instead of @IsNotEmpty

Reply:
Fixed